app/views/ProjectEdit/Framework/FrameworkDetail/index.tsx

```diff
@@ -90,7 +157,17 @@ function FrameworkDetail(props: Props) {
             analysisFramework: frameworkId,
         }),
         onSuccess: (response) => {
-            onProjectChange(response);
+            setProject((oldProjectDetails) => {
+                if (!oldProjectDetails) {
+                    return oldProjectDetails;
+                }
+                return ({
+                    ...oldProjectDetails,
+                    analysisFramework: {
+                        id: String(response.analysisFramework),
```

Comment:
Are we sure that `repsonse.analyticalFramework` is always a number and not undefined?

Reply:
Yes

Reply:
What if we sent undefined during project patch? Let's add proper check